packages/editor/src/store/reducer.js

```diff
@@ -97,6 +101,23 @@ function getFlattenedBlocks( blocks ) {
 	return flattenedBlocks;
 }
 
+/**
+ * Returns an object against which it is safe to perform mutating operations,
+ * given the original object and its current working copy.
+ *
+ * @param {Object} original Original object.
+ * @param {Object} working  Working object.
+ *
+ * @return {Object} Mutation-safe object.
+ */
+function getMutateSafeObject( original, working ) {
```

Comment:
Can we consider https://github.com/mweststrate/immer instead of this?

Reply:
I'm not too opposed to the idea of it as a _general_ pattern, but for one-off use it seems maybe a bit much both in terms of bundle size impact and developer knowledge barrier. Should we discuss separate to this pull request? I do think Immer seems to tackle well a pattern which is very prevalent in our stores.

Reply:
`getMutateSafeObject` can also be considered "something to learn" but yes definitely a topic for a separate discussion.

Reply:
There are degrees of learning 😆